pkg/analyze/data_source_test.go

```diff
@@ -0,0 +1,146 @@
+package analyze
+
+import (
+	"context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+	"reflect"
+	"testing"
+)
+
+type dummyLister struct {
+	indexer cache.Indexer
+}
+
+type dummyObject struct {
+	metav1.ObjectMeta
+	id int64
+}
+
+func (d dummyObject) GetObjectKind() schema.ObjectKind {
+	return schema.EmptyObjectKind
+}
+
+func (d dummyObject) DeepCopyObject() runtime.Object {
+	return d
+}
+
```

Comment:
please also add a test for `NewDataSourceFromClients`. You can use a fake client created from a set of objects

Reply:
done